### PR TITLE
feat(kafka): SASL authentication and additional configuration

### LINKED ISSUE #273: Kafka authentication/encryption
**Motivation**

I would like to use the Falcosidekick Kafka output to communicate with a public Kafka endpoint. This does require authentication and encryption to secure the connection between Falcosidekick and the Kafka endpoint.

**Feature**
Enable authentication and encryption for Kafka in Falcosidekick. Mutual TLS seems to be a good choice, as it has already been implemented for many Falcosidekick outputs and is supported by Kafka.

**Alternatives**

An alternative is to create a private tunnel between the nodes running Falcosidekick and Kafka. This adds additional complexity and is not a preferred solution.

### PR DESCRIPTION
**Pull request by lyoung-confluent**
**What type of PR is this?**
/kind feature
/area outputs

**What this PR does / why we need it**:
This PR adds some new features to [the Kafka output](./outputs/kafka.go):
* SASL authentication including PLAIN, SCRAM_SHA256 and SCRAM_SHA512
* Message compression as either GZIP, SNAPPY, LZ4, ZSTD or NONE (default)
* Partition balancing as crc32, murmer2, least_bytes and round_robin (default)
* Async message production via a completion callback
* Other additional configuration knobs like `required_acks` and `client.id`

All of these features already existed in [the latest version of the upstream Kafka library](https://pkg.go.dev/github.com/segmentio/kafka-go@v0.4.34) that was already in use, this is mostly just boilerplate to wire up the configuration.

**Which issue(s) this PR fixes**:
Fixes #273 (not including MTLS support)

### CODE REVIEW

File: outputs/kafka.go [hunk 1]
```diff
@@ -3,31 +3,119 @@ package outputs
...
 
 	"github.com/falcosecurity/falcosidekick/types"
 )
 
 // NewKafkaClient returns a new output.Client for accessing the Apache Kafka.
 func NewKafkaClient(config *types.Configuration, stats *types.Statistics, promStats *types.PromStatistics, statsdClient, dogstatsdClient *statsd.Client) (*Client, error) {
 
+	transport := &kafka.Transport{
+		Dial: (&net.Dialer{
+			Timeout:   3 * time.Second,
```
Comment (Issif) [inline outputs/kafka.go:25]: do we want to allow people to set this timeout?

File: outputs/kafka.go [hunk 2]
```diff
@@ -3,31 +3,119 @@ package outputs
...
+			case "SHA512":
+				transport.SASL, err = scram.Mechanism(scram.SHA512, config.Kafka.Username, config.Kafka.Password)
+			default:
+				return nil, fmt.Errorf("unsupported SASL SCRAM algorithm %q", algo)
+			}
+			if err != nil {
+				return nil, fmt.Errorf("failed to initialize SASL SCRAM %q: %w", algo, err)
+			}
+		default:
+			return nil, fmt.Errorf("unsupported SASL mode: %q", config.Kafka.SASL)
```
Comment (Issif) [inline outputs/kafka.go:54]: Can you also print a log line before each error:
`log.Printf("[ERROR] : Kafka - %v\n", errors.New("")` (example)

File: outputs/kafka.go (hunk 1 above)
Comment (lyoung-confluent) [reply outputs/kafka.go:25]: Probably, this value is the same as what is was prior to this change:
```go
var DefaultTransport RoundTripper = &Transport{
	Dial: (&net.Dialer{
		Timeout:   3 * time.Second,
		DualStack: true,
	}).DialContext,
}
```
I wasn't sure about going overboard on making every possible value configurable

Comment (Issif) [reply outputs/kafka.go:25]: Ok, let's keep it like it is right now. Thanks

File: outputs/kafka.go (hunk 2 above)
Comment (Issif) [reply outputs/kafka.go:54]: up

**Comment by Issif** [comment]
The PR #385 is same than yours, I asked the author to co-sign you as author.

**Pull request closed without merging**